py-rattler/src/record.rs

```diff
@@ -0,0 +1,406 @@
+use std::path::PathBuf;
+
+use pyo3::{
+    exceptions::PyTypeError, intern, pyclass, pymethods, FromPyObject, PyAny, PyErr, PyResult,
+};
+use rattler_conda_types::{
+    package::{IndexJson, PackageFile},
+    PackageRecord, PrefixRecord, RepoDataRecord,
+};
+
+use rattler_digest::{parse_digest_from_hex, Md5, Sha256};
+
+use crate::{
+    error::PyRattlerError, package_name::PyPackageName, prefix_record::PyPrefixPaths,
+    version::PyVersion,
+};
+
+#[pyclass]
+#[repr(transparent)]
+#[derive(Clone)]
+pub struct PyRecord {
```

Comment:
Can you add some comments here that explain the structure of this type? With the enum etc. And why we do it this way?

Reply:
Please also explain why there is an `inner`.